PWGHF/TableProducer/HFTreeCreatorLcToPKPi.cxx

```diff
@@ -216,7 +219,8 @@ struct CandidateTreeWriter {
                            float FunctionCt,
                            float FunctionY,
                            float FunctionE) {
-        if (FunctionSelection >= 1) {
+        double pseudoRndm = trackPos1.pt() * 1000. - (long)(trackPos1.pt() * 1000);
+        if (FunctionSelection >= 1 && std::abs(candidate.flagMCMatchRec()) >= isSignal && pseudoRndm < downSampleBkgFactor) {
```

Comment:
This is wrong.
```c++
        std::abs(candidate.flagMCMatchRec()) >= isSignal
```
`flagMCMatchRec` returns a bitmap which will be >= 1 for all 3-prong decay channels.
You have to test the right bit if you want to select only Lc.

Reply:
Dear Vit, 
thanks for pointing this out. Indeed this check did not select the Lc signal. I added the test on the LctoPKpi in the selection now.